File: LeetCode/Normal/400-499/404.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sumOfLeftLeaves(self, root: TreeNode) -> int:
        number = 0

        def dfs(root: TreeNode, flag=0):
            if not root:
                return
            if not root.left and not root.right and flag:
                nonlocal number
                number += root.val
                return
            dfs(root.left, 1)
            dfs(root.right, 0)

        dfs(root, 0)
        return number

File: LeetCode/Normal/400-499/404.cpp
#include <bits/stdc++.h>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution
{
public:
    int sumOfLeftLeaves(TreeNode *root)
    {
        if (!root)
            return 0;
        int l = sumOfLeftLeaves(root->left), r = sumOfLeftLeaves(root->right);
        if (root->left && !root->left->left && !root->left->right)
            return l + r + root->left->val;
        return l + r;
    }
};